src/test/analyzer_test.ts

```diff
@@ -779,6 +779,41 @@ var DuplicateNamespace = {};
     });
   });
 
+  test('maintain caches across multiple edits', async() => {
+    const b1Doc = await analyzer.analyze('base.js', `
+        class BaseElement extends HTMLElement {}
+        customElements.define('base-elem', BaseElement);
+    `);
+    assert.deepEqual(b1Doc.getWarnings(), []);
+    const u1Doc = await analyzer.analyze('user.html', `
+        <script src="./base.js"></script>
+        <script>
+          class UserElem extends BaseElement {}
+          customElements.define('user-elem', UserElem);
+        </script>
+    `);
+    assert.deepEqual(u1Doc.getWarnings(), []);
+
+    const b2Doc = await analyzer.analyze('base.js', `
+        class VanEl extends HTMLElement {}
+        customElements.define('base-elem', VanEl);
+    `);
+    assert.deepEqual(b2Doc.getWarnings(), []);
+    const u2Doc = await analyzer.analyze('user.html');
+    assert.notEqual(u1Doc, u2Doc);
+    assert.equal(
+        u2Doc.getWarnings()[0].message,
+        'Unable to resolve superclass BaseElement');
+
+    const b3Doc = await analyzer.analyze('base.js', `
+        class BaseElement extends HTMLElement {}
+        customElements.define('base-elem', BaseElement);
+    `);
+    assert.deepEqual(b3Doc.getWarnings(), []);
+    const u3Doc = await analyzer.analyze('user.html');
+    assert.equal(u3Doc.getWarnings().length, 0);
+  });
```

Comment:
This is the test of the specific invalid behavior that inspired this work.

I added more specific tests for the issue in dependency-graph_test as well below.